doc/man7/EVP_PKEY-RSA.pod

```diff
@@ -132,6 +132,17 @@ The RSA "e" value. The value may be any odd number greater than or equal to
 65537. The default value is 65537.
 For legacy reasons a value of 3 is currently accepted but is deprecated.
 
+=item "rsa-derive-from-pq"  (B<OSSL_PKEY_PARAM_RSA_DERIVE_FROM_PQ>) <unsigned integer>
+
+Indicate that missing parameters not passed in the parameter list should be
+derived if not provided.  Setting a nonzero value will cause all
+needed exponents and coefficients to be derived if not available.  Setting this
+option requires at least OSSL_PARAM_RSA_FACTOR1, OSSL_PARAM_RSA_FACTOR2,
+and OSSL_PARAM_RSA_N to be provided.  If OSSL_KEYMGMT_SELECT_PRIVATE_KEY is set
+in the selection field, then OSSL_PARAM_RSA_D is also required.
```

Comment:
D is needed for this computation and the OSSL_PKEY_PARAM_RSA_DERIVE_FROM_PQ should be ignored if OSSL_KEYMGMT_SELECT_PRIVATE_KEY is not set.

Reply:
That doesn't seem right to me.  If a user indicates OSSL_PKEY_PARAM_RSA_DERIVE_FROM_PQ , the should then be required to provide the inputs needed to perform that operation (which implies the specification of D)  Ignoring a requested operation when not providing the needed inputs seems like it will be confusing to users.

Reply:
The other option is to error out if OSSL_PKEY_PARAM_RSA_DERIVE_FROM_PQ is set and OSSL_KEYMGMT_SELECT_PRIVATE_KEY is not set in the selection.

The current wording says that D is needed only if OSSL_KEYMGMT_SELECT_PRIVATE_KEY is set which is true but what is also true is that although the code checks whether P and Q is present it does not do anything with it in that case. Which is also logical because there is no point computing the CRT params if what we are importing is not a private key.

So either
1. Ignore the OSSL_PKEY_PARAM_RSA_DERIVE_FROM_PQ when OSSL_KEYMGMT_SELECT_PRIVATE_KEY is not set and do not do anything with P, Q or D, not even check its presence.
2. Or error out if OSSL_PKEY_PARAM_RSA_DERIVE_FROM_PQ is set and OSSL_KEYMGMT_SELECT_PRIVATE_KEY is not set (regardless of whether P, Q or D is present or not) because the request is to derive CRT params which are not part of public keys.

Currently this PR is doing something in the middle which is IMO incorrect.

Reply:
Ok, thank you, based on your more detailed explanation, I can see how ignoring DERIVE_FROM_PQ makes sense if OSSL_KEYMGMT_SELECT_PRIVATE_KEY isn't set, as it doesn't matter if p and q are present in that case.  I'll make the change shortly